smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/config/ConfigProperty.kt

```diff
@@ -6,19 +6,19 @@
 package software.amazon.smithy.swift.codegen.config
 
 import software.amazon.smithy.codegen.core.Symbol
+import software.amazon.smithy.swift.codegen.SwiftWriter
 import software.amazon.smithy.swift.codegen.model.isOptional
 
 data class ConfigProperty(
     val name: String,
     val type: Symbol,
     val default: DefaultProvider? = null
 ) {
-    internal val additionalTypes: Set<Symbol> = emptySet()
 
     constructor(
         name: String,
         type: Symbol,
-        default: String,
+        default: (SwiftWriter) -> String,
```

Comment:
`default` is changed to a closure that takes a SwiftWriter so that any symbols that are part of the default value can be properly recorded.